app/commands/update_current_month_datasource_expenses.py

```diff
@@ -93,13 +93,14 @@ async def store_datasource_expenses(
     for organization_id, datasources in datasources_per_organization_id.items():
         for datasource in datasources:
             existing_datasource_expense, created = await datasource_expense_handler.get_or_create(
-                datasource_id=datasource["id"],
+                linked_datasource_id=datasource["id"],
                 organization_id=organization_id,
                 year=year,
                 month=month,
                 defaults={
                     "month_expenses": datasource["details"]["cost"],
                     "datasource_name": datasource["name"],
+                    "datasource_id": datasource["account_id"],
```

Comment:
you should add the `linked_datasource_id` to default and, in any case the it should be updated together with the `month_expenses` field.